kafka/channel/pkg/dispatcher/dispatcher.go

```diff
@@ -231,90 +212,46 @@ func (d *KafkaDispatcher) subscribe(channelRef provisioners.ChannelReference, su
 	d.logger.Info("Subscribing", zap.Any("channelRef", channelRef), zap.Any("subscription", sub))
 
 	topicName := d.topicFunc(utils.KafkaChannelSeparator, channelRef.Namespace, channelRef.Name)
-	group := fmt.Sprintf("%s.%s", controller.Name, sub.UID)
-	consumer, err := d.kafkaCluster.NewConsumer(group, []string{topicName})
+	groupID := fmt.Sprintf("%s.%s", controller.Name, sub.UID)
+
+	handler := consumerMessageHandler{sub, d.dispatcher }
+
+	consumerGroup, err := d.kafkaConsumerFactory.StartConsumerGroup(groupID, []string{topicName}, d.logger, handler)
 
 	if err != nil {
 		// we can not create a consumer - logging that, with reason
 		d.logger.Info("Could not create proper consumer", zap.Error(err))
 		return err
 	}
 
-	channelMap, ok := d.kafkaConsumers[channelRef]
-	if !ok {
-		channelMap = make(map[subscription]KafkaConsumer)
-		d.kafkaConsumers[channelRef] = channelMap
-	}
-	channelMap[sub] = consumer
-
-	if cluster.ConsumerModePartitions == d.kafkaCluster.GetConsumerMode() {
-		go d.partitionConsumerLoop(consumer, channelRef, sub)
-	} else {
-		go d.multiplexConsumerLoop(consumer, channelRef, sub)
-	}
-	return nil
-}
-
-func (d *KafkaDispatcher) partitionConsumerLoop(consumer KafkaConsumer, channelRef provisioners.ChannelReference, sub subscription) {
-	d.logger.Info("Partition Consumer for subscription started", zap.Any("channelRef", channelRef), zap.Any("subscription", sub))
-	for {
-		pc, more := <-consumer.Partitions()
-		if !more {
-			break
+	// sarama reports error in consumerGroup.Error() channel
+	// this goroutine logs errors incoming
+	go func() {
+		for err = range consumerGroup.Errors() {
+			d.logger.Warn("Error in consumer group", zap.Error(err))
```

Comment:
Why is this warn level instead of error?

Reply:
I'm not sure all errors in this channel can be considered with log level errors honestly